docs/containers/debug-node.md

```diff
@@ -124,10 +115,107 @@ Also note that the `debug` logger will not write logs unless enabled via the `DE
+### Configuring when the application is "ready"
+
+The extension determines the application is "ready" to receive HTTP connections when it writes a message of the form `Listening on port <number>` to the debug console, as Express.js does by default.  If the application logs a different
+message, then the `pattern` property of the [`dockerServerReadyAction`]((debug-common.md#dockerServerReadyAction-object-properties)) object of the debug launch configuration should be set to a [JavaScript regular expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) that matches that message. The regular expression should include a capture group that corresponds to the port on which the application is listening.
+
+For example, suppose the application logs the following message:
+
+```js
+function onListening() {
+  var addr = server.address();
+  var bind = typeof addr === 'string'
+    ? 'pipe ' + addr
+    : 'port ' + addr.port;
+  debug('Application has started on ' + bind);
+}
+```
+
+The corresponding `pattern` in the debug launch configuration (in `launch.json`) would be:
+
+```json
+{
+    "configurations": [
+        {
+            "name": "Docker Node.js Launch and Attach",
+            "type": "docker",
+            "request": "launch",
+            "preLaunchTask": "docker-run: debug",
+            "platform": "node",
+            "dockerServerReadyAction": {
+                "pattern": "Application has started on port (\\d+)"
+            }
+        }
+    ]
+}
+```
+
+### Configuring the application entry page
+
+By default, the Docker extension will open the "main" page of the browser (however that is determined by the application).  If the browser should be opened to a specific page, the `uriFormat` property of the [`dockerServerReadyAction`]((debug-common.md#dockerServerReadyAction-object-properties)) object of the debug launch configuration should be set to a Node.js format string, with one string token that indicates where the port should be substituted.
+
+The corresponding `uriFormat` in the debug launch configuration (in `launch.json`) to open the `about.html` page instead of the main page would be:
+
+```json
+{
+    "configurations": [
+        {
+            "name": "Docker Node.js Launch and Attach",
+            "type": "docker",
+            "request": "launch",
+            "preLaunchTask": "docker-run: debug",
+            "platform": "node",
+            "dockerServerReadyAction": {
+                "uriFormat": "http://localhost:%s/about.html"
+            }
+        }
+    ]
+}
+```
+
+## Mapping Docker container source files to the local workspace
+
+By default, the Docker extension assumes the application source files in the running Docker container are located in an `/usr/src/app` folder, and the debugger then maps those files back to the root of the opened workspace, in order to translate breakpoints from the container back to Visual Studio Code.
```

Comment:
Are these paths Linux-specific?  Is the Windows version relevant here?

Reply:
We scaffold only Linux container Dockerfiles for Node.js applications; to our knowledge, using Windows containers to host Node.js applications is not a common scenario.  (Note that Node.js is often installed on Windows containers targeting .NET Core applications as Node.js is often part of the build toolchain for the JavaScript front-ends hosted by .NET Core web applications.)

Even for local paths, when the user is using VS Code on Windows, the paths are typically relative to a `${workspaceFolder}` token, in which case Linux-style paths are still valid.